src/builder/build_traits.rs

```diff
@@ -29,17 +30,40 @@ use crate::Hugr;
 
 use crate::hugr::HugrMut;
 
+pub trait Buildable {
+    type BaseMut<'a>: Buildable + HugrMut
```

Comment:
We could avoid this `Buildable + HugrMut` everywhere if we wrote a blanket `impl<B:Buildable> HugrMut for B` (just by delegating to `self.hugr_mut()`). That would let us just constrain everything by `Buildable` instead, which would be nicer. OTOH it means one can execute HugrMut methods on every Buildable, of which there are many (every Container, etc.)....maybe that'd be a good thing (and could be done as a follow-up PR if we want to see what it looks like?)